test/ngAnimate/animateCacheSpec.js

```diff
@@ -0,0 +1,96 @@
+  }));
+
+  it('should update existing stored details in a lookup', inject(function($$animateCache) {
+    var data = { 'hello': 'there' };
+    $$animateCache.put('key', data, true);
+
+    var otherData = { 'hi': 'you' };
+    $$animateCache.put('key', otherData, true);
+    expect($$animateCache.get('key')).toBe(otherData);
+  }));
+
+  it('should create a special cacheKey based on the element/parent and className relationship', inject(function($$animateCache) {
+    var cacheKey, elm = jqLite('<div></div>');
+    elm.addClass('one two');
+
+    var parent1 = jqLite('<div></div>');
+    parent1.append(elm);
+
+    cacheKey = $$animateCache.cacheKey(getDomNode(elm), 'event');
+    expect(cacheKey).toBe('1 event one two');
+
+    cacheKey = $$animateCache.cacheKey(getDomNode(elm), 'event', 'add');
+    expect(cacheKey).toBe('1 event one two add');
+
+    cacheKey = $$animateCache.cacheKey(getDomNode(elm), 'event', 'add', 'remove');
+    expect(cacheKey).toBe('1 event one two add remove');
+
+    var parent2 = jqLite('<div></div>');
+    parent2.append(elm);
+
+    cacheKey = $$animateCache.cacheKey(getDomNode(elm), 'event');
+    expect(cacheKey).toBe('2 event one two');
+
+    cacheKey = $$animateCache.cacheKey(getDomNode(elm), 'event', 'three', 'four');
+    expect(cacheKey).toBe('2 event one two three four');
+  }));
+
+  it('should keep a count of how many times a cache key has been updated', inject(function($$animateCache) {
+    var data = { 'hello': 'there' };
+    var key = 'key';
+    expect($$animateCache.count(key)).toBe(0);
+
+    $$animateCache.put(key, data, true);
+    expect($$animateCache.count(key)).toBe(1);
+
+    var otherData = { 'other': 'data' };
+    $$animateCache.put(key, otherData, true);
+    expect($$animateCache.count(key)).toBe(2);
+  }));
+
+  it('should flush the cache and the counters', inject(function($$animateCache) {
+    $$animateCache.put('key1', { data: 'value' }, true);
+    $$animateCache.put('key2', { data: 'value' }, true);
+
+    expect($$animateCache.count('key1')).toBe(1);
+    expect($$animateCache.count('key2')).toBe(1);
+
+    $$animateCache.flush();
+
+    expect($$animateCache.get('key1')).toBeFalsy();
+    expect($$animateCache.get('key2')).toBeFalsy();
+
+    expect($$animateCache.count('key1')).toBe(0);
+    expect($$animateCache.count('key2')).toBe(0);
+  }));
+
+  it('should store whether the stored entry is invalid or not', inject(function($$animateCache) {
+    var validEntry = { someEssentialProperty: true };
+    var invalidEntry = { someEssentialProperty: false };
+
+    $$animateCache.put('key1', validEntry, true);
+    $$animateCache.put('key2', invalidEntry, false);
+
+    expect($$animateCache.containsCachedValidAnimation('key1')).toBe(true);
+    expect($$animateCache.containsCachedValidAnimation('key2')).toBe(false);
+  }));
+
+  it('should consider a cached entry valid if it does not exist in the cache', inject(function($$animateCache) {
```

Comment:
Do you mena `It should consider a key valid if it does not exist in the cache`?  Because a cached entry that doesn't exist in the cache is not possible.